### PR TITLE
Only call refresh when the locale is different in the entity

### LINKED ISSUE #506: Bug on OneToMany CollectionType with Translations
#### Sonata packages

<details><summary>show</summary>
<p>

```
sonata-project/admin-bundle              3.104.0 3.104.0 The missing Symfony...
sonata-project/block-bundle              3.23.2  4.6.0   Symfony SonataBlock...
sonata-project/cache                     2.0.1   2.1.1   Cache library
sonata-project/core-bundle               3.20.0  3.20.0  Symfony SonataCoreB...
Package sonata-project/core-bundle is abandoned, you should avoid using it. No replacement was suggested.
sonata-project/datagrid-bundle           3.3.0   3.3.0   Symfony SonataDatag...
sonata-project/doctrine-extensions       1.13.1  1.13.1  Doctrine2 behaviora...
sonata-project/doctrine-orm-admin-bundle 3.35.0  3.35.0  Integrate Doctrine ...
sonata-project/exporter                  2.7.0   2.7.0   Lightweight Exporte...
sonata-project/form-extensions           0.1.2   1.9.0   Symfony form extens...
sonata-project/formatter-bundle          4.5.0   4.5.0   Symfony SonataForma...
sonata-project/intl-bundle               2.10.1  2.10.1  Symfony SonataIntlB...
sonata-project/media-bundle              3.32.0  3.32.0  Symfony SonataMedia...
sonata-project/seo-bundle                2.13.1  2.13.1  Symfony SonataSeoBu...
sonata-project/translation-bundle        2.8.1   2.8.1   SonataTranslationBu...
sonata-project/twig-extensions           0.1.1   1.7.0   Sonata twig extensions
```

</p>
</details>

#### Symfony packages

<details><summary>show</summary>
<p>

```
symfony/asset                      v4.4.27 v5.3.4  Manages URL generation an...
symfony/cache                      v4.4.27 v5.3.4  Provides an extended PSR-...
symfony/cache-contracts            v2.4.0  v2.4.0  Generic abstractions rela...
symfony/config                     v4.4.27 v5.3.4  Helps you find, load, com...
symfony/console                    v4.4.27 v5.3.4  Eases the creation of bea...
symfony/css-selector               v4.4.27 v5.3.4  Converts CSS selectors to...
symfony/debug                      v4.4.27 v4.4.27 Provides tools to ease de...
symfony/dependency-injection       v4.4.27 v5.3.4  Allows you to standardize...
symfony/deprecation-contracts      v2.4.0  v2.4.0  A generic function and co...
symfony/doctrine-bridge            v4.4.27 v5.3.4  Provides integration for ...
symfony/dotenv                     v4.4.27 v5.3.4  Registers environment var...
symfony/error-handler              v4.4.27 v5.3.4  Provides tools to manage ...
symfony/event-dispatcher           v4.4.27 v5.3.4  Provides tools that allow...
symfony/event-dispatcher-contracts v1.1.9  v2.4.0  Generic abstractions rela...
symfony/expression-language        v4.4.27 v5.3.4  Provides an engine that c...
symfony/filesystem                 v4.4.27 v5.3.4  Provides basic utilities ...
symfony/finder                     v4.4.27 v5.3.4  Finds files and directori...
symfony/flex                       v1.13.3 v1.13.3 Composer plugin for Symfony
symfony/form                       v4.4.27 v5.3.4  Allows to easily create, ...
symfony/framework-bundle           v4.4.27 v5.3.4  Provides a tight integrat...
symfony/http-client                v4.4.27 v5.3.4  Provides powerful methods...
symfony/http-client-contracts      v2.4.0  v2.4.0  Generic abstractions rela...
symfony/http-foundation            v4.4.27 v5.3.4  Defines an object-oriente...
symfony/http-kernel                v4.4.28 v5.3.5  Provides a structured pro...
symfony/inflector                  v4.4.27 v5.3.4  Converts words between th...
symfony/intl                       v4.4.27 v5.3.4  Provides a PHP replacemen...
symfony/mailer                     v4.4.27 v5.3.4  Helps sending emails
symfony/mailgun-mailer             v4.4.27 v5.3.4  Symfony Mailgun Mailer Br...
symfony/mime                       v4.4.27 v5.3.4  Allows manipulating MIME ...
symfony/monolog-bridge             v4.4.27 v5.3.4  Provides integration for ...
symfony/monolog-bundle             v3.7.0  v3.7.0  Symfony MonologBundle
symfony/options-resolver           v4.4.27 v5.3.4  Provides an improved repl...
symfony/phpunit-bridge             v4.4.27 v5.3.4  Provides utilities for PH...
symfony/polyfill-intl-grapheme     v1.23.0 v1.23.0 Symfony polyfill for intl...
symfony/polyfill-intl-icu          v1.23.0 v1.23.0 Symfony polyfill for intl...
symfony/polyfill-intl-idn          v1.23.0 v1.23.0 Symfony polyfill for intl...
symfony/polyfill-intl-normalizer   v1.23.0 v1.23.0 Symfony polyfill for intl...
symfony/polyfill-mbstring          v1.23.0 v1.23.0 Symfony polyfill for the ...
symfony/polyfill-php72             v1.23.0 v1.23.0 Symfony polyfill backport...
symfony/polyfill-php73             v1.23.0 v1.23.0 Symfony polyfill backport...
symfony/polyfill-php80             v1.23.0 v1.23.0 Symfony polyfill backport...
symfony/polyfill-php81             v1.23.0 v1.23.0 Symfony polyfill backport...
symfony/polyfill-uuid              v1.23.0 v1.23.0 Symfony polyfill for uuid...
symfony/process                    v4.4.27 v5.3.4  Executes commands in sub-...
symfony/property-access            v4.4.27 v5.3.4  Provides functions to rea...
symfony/routing                    v4.4.27 v5.3.4  Maps an HTTP request to a...
symfony/security-acl               v3.2.0  v3.2.0  Symfony Security Componen...
symfony/security-bundle            v4.4.27 v5.3.4  Provides a tight integrat...
symfony/security-core              v4.4.27 v5.3.4  Symfony Security Componen...
symfony/security-csrf              v4.4.27 v5.3.4  Symfony Security Componen...
symfony/security-guard             v4.4.27 v5.3.4  Symfony Security Componen...
symfony/security-http              v4.4.27 v5.3.4  Symfony Security Componen...
symfony/serializer                 v4.4.27 v5.3.4  Handles serializing and d...
symfony/service-contracts          v2.4.0  v2.4.0  Generic abstractions rela...
symfony/stopwatch                  v4.4.27 v5.3.4  Provides a way to profile...
symfony/string                     v5.3.3  v5.3.3  Provides an object-orient...
symfony/templating                 v4.4.27 v5.3.4  Provides all the tools ne...
symfony/translation                v4.4.27 v5.3.4  Provides tools to interna...
symfony/translation-contracts      v2.4.0  v2.4.0  Generic abstractions rela...
symfony/twig-bridge                v4.4.27 v5.3.4  Provides integration for ...
symfony/twig-bundle                v4.4.27 v5.3.4  Provides a tight integrat...
symfony/validator                  v4.4.27 v5.3.4  Provides tools to validat...
symfony/var-dumper                 v4.4.27 v5.3.4  Provides mechanisms for w...
symfony/var-exporter               v4.4.27 v5.3.4  Allows exporting any seri...
symfony/web-profiler-bundle        v4.4.28 v5.3.5  Provides a development to...
symfony/web-server-bundle          v4.4.27 v4.4.27 Provides commands for run...
symfony/webpack-encore-bundle      v1.12.0 v1.12.0 Integration with your Sym...
symfony/yaml                       v4.4.27 v5.3.4  Loads and dumps YAML files

```

</p>
</details>

#### PHP version

```
PHP 7.4.20 (cli) (built: Jun  4 2021 21:24:55) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.20, Copyright (c), by Zend Technologies

```

## Subject

I have a Product entity with an attributes field (OneToMany) pointing to a collection of ProductAttribute entity. 
Both entities are translatables. 

I'm declaring a CollectionType in my ProductAdmin class

```
$formMapper->add(
                "attributes",
                CollectionType::class,
                [
                    'label'        => false,
                    'by_reference' => false,
                    'required'     => false,
                ],
                [
                    'edit'     => 'inline',
                    'inline'   => 'table',
                    'sortable' => 'position',
                ]
            )
```

## Expected results

On the Product administration page : 
- I click on the add new button of the attributes table
- I fill my new line
- I click again on the add new button
- I have a new blank line to fill

## Actual results

On the Product administration page : 
- I click on the add new button of the attributes table
- I fill my new line
- I click again on the add new button
- **It resets my first line i have filled**

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Try to fix https://github.com/sonata-project/SonataTranslationBundle/issues/506

I've never had this problem because I'm not using the bundle with an admin with sub items, this is just a try to see if it works.

@benjamin-hubert if you could please try it.

@wokenlex I don't know if your problem is related also with this.

It's only changed when using non-deprecated interfaces (only implementing `Translatable` from Gedmo and not `TranslatableInterface` from this bundle).

I am targeting this branch, because it is a bugfix.

Closes #506

## Changelog

```markdown
### Fixed
- Creating or updating an entity with translatable sub items
```

If this is fine, I would try to create a test with panther

### CODE REVIEW

**Comment by wokenlex** [comment]
>I don't know if your problem is related also with this.

Probably it is. I has removed the hack and added you patch - it's look like it works, will need to make more tests.

**Comment by VincentLanglet** [comment]
@franmomu does this PR ready ?

@wokenlex Did you have time for more tests ?

**Comment by franmomu** [comment]
I'll try later to at least add a unit test

File: tests/Admin/Extension/Gedmo/TranslatableAdminExtensionTest.php [hunk 1]
```diff
@@ -126,6 +125,44 @@ public function testConfigureQuery(): void
         static::assertFalse($this->translatableListener->getTranslationFallback());
     }
 
+    /**
+     * @psalm-suppress InvalidArgument Each extension will handle specific type on NEXT_MAJOR
```
Comment (VincentLanglet) [inline tests/Admin/Extension/Gedmo/TranslatableAdminExtensionTest.php]: I'm not certain to understand, does this need to be updated in NEXT_MAJOR ?

Comment (franmomu) [reply tests/Admin/Extension/Gedmo/TranslatableAdminExtensionTest.php]: I've changed it

**Review by VincentLanglet — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thanks @franmomu

**Comment by wokenlex** [comment]
…After upgrade to sonata admin bundle - this problem returns back. Im saving in one locale, it saves everywhere and destroyers content.

**Pull request merged**